students/Dustin_L/lesson03/mailroom.py

```diff
@@ -0,0 +1,179 @@
+    three, that value is returned. If not, the user is prompted again to select.
+
+    Returns:
+        int: Value corresponding to user choice
+    """
+    select_prompt = '\nPlease select from the following options:\n'\
+                    '\t1. Send a Thank You\n'\
+                    '\t2. Create a Report\n'\
+                    '\t3. quit\n'\
+                    ' --> '
+    while True:
+        usr_in = int(input(select_prompt).strip())
+
+        if usr_in == THANK_YOU_OPT or usr_in == REPORT_OPT or usr_in == QUIT_OPT:
+            break
+        else:
+            print('\nPlease enter either a "1", "2", or "3"\n')
+
+    return usr_in
+
+
+def add_donation(idx, amount):
+    """Add a new donation to the donation database.
+
+    Args:
+        idx (int): Index of donor in database.
+        amount (int): Amount to add to donation database.
+    """
+    DONATION_DB[idx][GIFTS_IDX].append(amount)
+    DONATION_DB[idx][NUM_GIFTS_IDX] += 1
+    DONATION_DB[idx][TOTAL_IDX]     += amount
+    DONATION_DB[idx][AVE_IDX]        = DONATION_DB[idx][TOTAL_IDX] /    \
+                                       DONATION_DB[idx][NUM_GIFTS_IDX]
+
+
+def send_thank_you():
+    """Send a thank you.
+
+    Prompt for a Full Name.
+    If the user types ‘list’, show them a list of the donor names and re-prompt
+    If the user types a name not in the list, add that name to the data structure
+    and use it.
+    If the user types a name in the list, use it.
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to
+    crash if someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of
+    the selected user.
+    Finally, use string formatting to compose an email thanking the donor for their
+    generous donation. Print the email to the terminal and return to the original prompt.
+    """
+    name_prompt   = '\nPlease enter name of "Thank You" recipient:\n'\
+                    '(Enter "list" to see all donors)\n'\
+                    '(Enter "quit" to return to main menu)\n'\
+                    ' --> '
+    amount_prompt = '\nPlease enter the donation amount:\n'\
+                    '(Enter "quit" to return to main menu)\n'\
+                    ' --> '
+    thank_you_fmt = '\nThank you {:s} for your generous donation of ${:.2f}!'
+    first_names   = [donor[NAME_IDX].lower().split()[NAME_IDX] for donor in DONATION_DB]
+
+    while True:
+        new_donor = False
+        usr_in    = input(name_prompt).strip().lower()
+
+        if usr_in.startswith('q'):
+            break
+        elif usr_in == 'list':
+            print()
+            for dnr in DONATION_DB:
+                print(dnr[NAME_IDX])
+        else:
+            if usr_in in first_names:
+                for i, _ in enumerate(DONATION_DB):
+                    if usr_in in DONATION_DB[i][NAME_IDX].lower():
```

Comment:
instead of using `DONATION_DB[i]` use value that would go into `_` that is already provided by `enumerate`